cstore_reader.c

```diff
@@ -1287,76 +1354,78 @@ ColumnDefaultValue(TupleConstr *tupleConstraints, Form_pg_attribute attributeFor
 }
 
 
-/* Returns the size of the given file handle. */
-static int64
-FileSize(FILE *file)
+/* ReadFromFile reads the given segment from the relation data file. */
+static StringInfo
+ReadFromFile(Relation relation, uint64 offset, uint32 size)
 {
-	int64 fileSize = 0;
-	int fseekResult = 0;
+	int blockCapacity = CSTORE_PAGE_DATA_SIZE;
+	uint32 blockNumber = offset / blockCapacity;
+	uint32 blockOffset = offset % blockCapacity;
+	uint32 remainingSize = size;
+	uint32 resultOffset = 0;
+	uint32 blockCount = 0;
+	StringInfo resultBuffer = NULL;
 
-	errno = 0;
-	fseekResult = fseeko(file, 0, SEEK_END);
-	if (fseekResult != 0)
+	if (size == 0)
```

Comment:
Gah, this is one of these cases where returning a variable is a lot less clear than just returning `NULL`.